src/cartridge/include/cartridge/meta_cartridge.h

```diff
@@ -0,0 +1,86 @@
+/**
+ * @file meta_cartridge.h
+ * Declares the MetaCartridge class
+ */
+
+#include "cartridge/utils.h"
+
+#include <cstdint>
+#include <string>
+#include <vector>
+
+#pragma once
+
+namespace cartridge {
+
+class MetaCartridge {
```

Comment:
Can this class have a better name? Something like `CartridgeMetadata` would be better. Also, you can add a docstring comment for the class itself too.